ui-test/src/main/java/pages/SignupFormDynamicFiller.java

```diff
@@ -0,0 +1,127 @@
+package pages;
+
+import org.apache.log4j.Logger;
+import org.openqa.selenium.*;
+import org.openqa.selenium.support.ui.Select;
+
+import utils.EsignetUtil;
+import utils.EsignetUtil.RegisteredDetails;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+
+public class SignupFormDynamicFiller {
+
+	private WebDriver driver;
+	private static final Logger logger = Logger.getLogger(SignupFormDynamicFiller.class);
+	RegistrationPage registrationPage;
+
+	public SignupFormDynamicFiller(WebDriver driver) {
+		this.driver = driver;
+		registrationPage = new RegistrationPage(driver);
+	}
+
+	public void fillFormFromUiSpec(Map<String, Map<String, Object>> uiSpecFields) {
+
+		for (String fieldId : uiSpecFields.keySet()) {
+
+			if (fieldId.equalsIgnoreCase("phone")) {
+				continue;
+			}
+
+			List<WebElement> matchingElements = driver
+					.findElements(By.xpath("//*[@id='" + fieldId + "' or @data-field-id='" + fieldId + "']"));
+
+			if (matchingElements.isEmpty()) {
+				logger.info("No element found for fieldId: " + fieldId);
+				continue;
+			}
+
+			WebElement element = matchingElements.get(0);
+			String tag = element.getTagName();
+			String type = element.getAttribute("type");
+
+			if (fieldId.equalsIgnoreCase("individualBiometrics")) {
```

Comment:
If its signup is running with mock then the biometric field id will be different. Instead of fieldId i think you should check with fieldType/controlType